schemas/manifest-schema.json

```diff
@@ -0,0 +1,376 @@
+        "then": {
+          "properties": {
+            "updatePolicy": {
+              "description": "A module's updatePolicy\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-updatepolicy",
+              "enum": [
+                "set",
+                "set_if_not_exists",
+                "append",
+                "add",
+                "min",
+                "max"
+              ]
+            },
+            "valueType": {
+              "description": "A module's valueType\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-valuetype",
+              "type": ["string", "number"]
+            },
+            "name": {
+              "description": "A module name\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-name",
+              "type": "string"
+            },
+            "initialBlock": {
+              "description": "A module initialBlock\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-initialBlock",
+              "type": "number"
+            },
+            "kind": {
+              "description": "A module kind\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-kind",
+              "enum": [
+                "map",
+                "store"
+              ]
+            },
+            "binary": {
+              "description": "A module binary\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-binary",
+              "type": "string"
+            },
+            "inputs": {
+              "description": "A module input\nhttps://substreams.streamingfast.io/reference-and-specs/manifests#module-inputs",
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "store": {
+                    "description": "an input store",
+                    "type": "string"
+                  },
+                  "source": {
+                    "description": "an input source",
+                    "type": "string"
+                  },
+                  "map": {
+                    "description": "an input map",
+                    "type": "string"
+                  }
+                },
+                "if": {
+                  "properties": {
+                    "store" : {
+                      "type": "string"
+                    }
+                  },
+                  "required": ["store"]
+                },
+                "then" : {
+                  "maxProperties": 2,
+                  "properties": {
+                    "mode": {
+                      "description": "an input mode",
+                      "type": "string"
+                    }
+                  }
+                },
+                "else": {
+                  "maxProperties": 1
+                },
+                "minProperties": 1,
+                "unevaluatedProperties": false
+              },
```

Comment:
I would have thought that the `oneOf` definition would have been clearer here no?

Reply:
not with the conditional property: mode, I ran into issues

Reply:
I think it will actually make your life easier because `mode` will be part of one of the definition, let's discuss this tomorrow.